src/resources/tls.cc

```diff
@@ -645,6 +650,60 @@ PRIMITIVE(close) {
   return process->null_object();
 }
 
+#define NEEDS_DELETE 1
```

Comment:
Not a fan of defines.
either make it a constant, an enum, or just pass the three values instead of the combined flag.